test/dependency_test.cr

```diff
@@ -5,25 +5,25 @@ class Shards::DependencyTest < Minitest::Test
     dependency = Dependency.new("app")
     assert_equal "*", dependency.version
 
-    dependency = Dependency.new("app", {version: "*"})
+    dependency = Dependency.new("app", version: "*")
```

Comment:
These unit tests are the only use case for an initializer with additional values. They don't make much sense anymore... unless we rewrite them a little:

```crystal
def test_version
  dependency = Dependency.new("app")
  assert_nil dependency.version?
  assert_equal "*", dependency.version

  dependency.version = "<= 2.0.0"
  assert_equal "<= 2.0.0", dependency.version?
  assert_equal "<= 2.0.0", dependency.version
end

def test_version_tags
  dependency = Dependency.new("app")

  dependency.tag = "fix/something"
  assert_equal "*", dependency.version

  dependency.tag = "v1.2.3-pre1"
  assert_equal "v1.2.3-pre1", dependency.version
end
```